### PR TITLE
Add support for backdrop filters in the compositor

### PR DESCRIPTION
**Pull request by abarth**
Particularly, add support for blur filters to implement backdrop blur effects.

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde @Hixie

File: sky/engine/core/dart/compositing.dart [hunk 1]
```diff
@@ -81,6 +81,14 @@ class SceneBuilder extends NativeFieldWrapperClass2 {
   /// See [pop] for details about the operation stack.
   void pushColorFilter(Color color, TransferMode transferMode) native "SceneBuilder_pushColorFilter";
 
+  /// Pushes a backdrop filter operation onto the operation stack.
```
Comment (Hixie) [inline sky/engine/core/dart/compositing.dart]: maybe elaborate on what a "backdrop filter" is?

File: sky/engine/core/dart/painting.dart [hunk 2]
```diff
@@ -214,6 +214,38 @@ class ColorFilter extends NativeFieldWrapperClass2 {
   void _constructor(Color color, TransferMode transferMode) native "ColorFilter_constructor";
 }
 
+/// A chain of filter operations to apply to an image.
```
Comment (Hixie) [inline sky/engine/core/dart/painting.dart]: image, or picture? In our world, they are quite distinct.

File: sky/engine/core/dart/painting.dart [hunk 3]
```diff
@@ -214,6 +214,38 @@ class ColorFilter extends NativeFieldWrapperClass2 {
...
+///
+/// See [SceneBuilder.pushBackdropFilter].
+class ImageFilter extends NativeFieldWrapperClass2 {
+  void _constructor() native "Gradient_constructor";
```
Comment (Hixie) [inline sky/engine/core/dart/painting.dart]: copypasta

File: flow/layers/backdrop_filter_layer.cc [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+void BackdropFilterLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
+  ContainerLayer::Preroll(context, matrix);
+  set_paint_bounds(context->child_paint_bounds);
+}
+
+void BackdropFilterLayer::Paint(PaintContext::ScopedFrame& frame) {
+  SkCanvas& canvas = frame.canvas();
+  SkAutoCanvasRestore save(&canvas, false);
+  canvas.saveLayer({&paint_bounds(), nullptr, filter_.get(), 0});
```
Comment (chinmaygarde) [inline flow/layers/backdrop_filter_layer.cc]: Minor: Can this say `canvas.saveLayer(SaveLayerRec{&paint_bounds(), nullptr, filter_.get(), 0})`. It is not immediately clear which overload you are using by just looking at the signature.

File: sky/engine/core/dart/painting.dart (hunk 2 above)
Comment (Hixie) [inline sky/engine/core/dart/painting.dart]: How does this relate to ImageShader?

File: sky/engine/core/dart/painting.dart (hunk 3 above)
Comment (chinmaygarde) [inline sky/engine/core/dart/painting.dart]: Rename to `ImageFilter_constructor`

**Comment by chinmaygarde** [comment]
lgtm with comments.

File: flow/layers/backdrop_filter_layer.cc (hunk 4 above)
Comment (abarth) [inline flow/layers/backdrop_filter_layer.cc]: Sure

**Pull request merged**